pydss/flapper.py

```diff
@@ -89,12 +89,12 @@ def kick(self, lot, bid) :
         require(self.kicks < -1,"Flapper/overflow")
 
         self.kicks+=1
-        self.bid_id += self.kicks;
+        bid_id = self.kicks;
 
-        self.bids[bid].bid = bid;
-        self.bids[self.bid_id].lot = lot;
-        self.bids[self.bid_id].guy = self.message.sender;
-        self.bids[self.bid_id].end = self.now+self.tau
+        self.bids[bid_id].bid = bid;
+        self.bids[bid_id].lot = lot;
+        self.bids[bid_id].guy = self.msg.sender;
+        self.bids[bid_id].end = self.now+self.tau
```

Comment:
`now` won't be an instance variable on `self`, you should pass it in as a parameter to the function